mypy/checker.py

```diff
@@ -1705,7 +1733,7 @@ def check_return_stmt(self, s: ReturnStmt) -> None:
                         self.warn(messages.RETURN_ANY.format(return_type), s)
                     return
 
-                if self.is_unusable_type(return_type):
+                if self.is_unusable_type(return_type) or defn.is_async_generator:
```

Comment:
I would prefer a separate (more clear) error message for async generators, like: "return statement with value in async generator is prohibited". Maybe it could be even blocking (since formally speaking this is a ``SyntaxError``).

Also, I would probably move this check to ``semanal.py``.